Autopilot/Src/Interchip_A.cpp

```diff
@@ -8,51 +8,66 @@ static volatile bool dataNew;
 
 // freertos run function
 void Interchip_Run() {
-	HAL_GPIO_WritePin(GPIOA, GPIO_PIN_4, GPIO_PIN_RESET);
-	HAL_StatusTypeDef result = HAL_SPI_TransmitReceive_IT(&hspi6,(uint8_t *)&txData,(uint8_t *)&rxData, 26);
+    HAL_GPIO_WritePin(GPIOA, GPIO_PIN_4, GPIO_PIN_RESET);
+
+    // I will put this into a function later - anni
+    // for now this is a quick way to compute crc and send it on our run.
+    int data = 0;
+    for(int i=0; i<12; i++) {
+        data += txData.PWM[i];
+    }
+    txData.crc = Interchip_CRC_Compute(data);
+
+    HAL_StatusTypeDef result = HAL_SPI_TransmitReceive_IT(&hspi6, (uint8_t * ) & txData, (uint8_t * ) & rxData, 26);
 }
 
 void Interchip_Init() {
-  HAL_GPIO_WritePin(GPIOA, GPIO_PIN_4, GPIO_PIN_SET);
-  // the rest of this program is just for like,
-  txData.safetyLevel = 14;
-  txData.PWM[0] = 0;
-	txData.PWM[1] = 0;
-	txData.PWM[2] = 0;
-	txData.PWM[3] = 0;
-	txData.PWM[4] = 0;
-	txData.PWM[5] = 0;
-	txData.PWM[6] = 0;
-	txData.PWM[7] = 0;
-	txData.PWM[8] = 0;
-	txData.PWM[9] = 0;
-	txData.PWM[10] = 0;
-	txData.PWM[11] = 0;
+    HAL_GPIO_WritePin(GPIOA, GPIO_PIN_4, GPIO_PIN_SET);
+    // the rest of this program is just for like,
+    txData.safetyLevel = 14;
```

Comment:
I think this stuff was just for testing, Im fairly sure you can just delete the txData.safetyLevel = and the txData.PWM[x] =